bin/reth/src/debug_cmd/merkle.rs

```diff
@@ -111,21 +118,22 @@ impl Command {
                     None
                 };
 
-            execution_stage
-                .execute(
-                    &mut tx,
-                    ExecInput {
-                        previous_stage: Some((StageId::SenderRecovery, block)),
-                        checkpoint: block.checked_sub(1).map(StageCheckpoint::new),
-                    },
-                )
-                .await?;
+            <ExecutionStage<Factory> as reth_stages::Stage<Arc<Env<WriteMap>>>>::execute(
+                &mut execution_stage,
+                &mut provider_rw,
```

Comment:
Hmm this is pretty ugly, why can't we do `executino_stage.execute::<Env<WriteMap>>` or something?